pkgs/development/tools/build-managers/bazel/python-stub-path-fix.patch

```diff
@@ -1,13 +0,0 @@
-diff --git a/src/main/java/com/google/devtools/build/lib/bazel/rules/python/python_stub_template.txt b/src/main/java/com/google/devtools/build/lib/bazel/rules/python/python_stub_template.txt
-index dac21c9a83..69b11c283f 100644
---- a/src/main/java/com/google/devtools/build/lib/bazel/rules/python/python_stub_template.txt
-+++ b/src/main/java/com/google/devtools/build/lib/bazel/rules/python/python_stub_template.txt
-@@ -67,7 +67,7 @@ def FindPythonBinary(module_space):
-     return os.path.join(module_space, PYTHON_BINARY)
-   else:
-     # Case 4: Path has to be looked up in the search path.
--    return SearchPath(PYTHON_BINARY)
-+    return "NIX_STORE_PYTHON_PATH"
```

Comment:
Why was this removed? Or rather, why was this removed, but the shebang line substitution was left in? Both force the built Bazel to hardcode the nix version of python in the built python binary. I'm not sure that was intentional, and I feel like that should be part of `enableNixHacks` instead of part of the regular build, but I don't think we should change this as part of this PR.

Reply:
The shebang line substitution is for the builder script itself which finds an appropriate Python binary; it's reasonable to pin this one as part of a usual build. On the other hand `NIX_STORE_PYTHON_PATH` was used to pin the Python binary and this is problematic for Python 2 builds.

Reply:
See also: https://github.com/NixOS/nixpkgs/pull/64903

Reply:
I see, so it's less about the nix version or the system version, and more about 2 vs 3. Ok, I have no issues with it then!